lib/src/blockdev.rs

```diff
@@ -50,6 +54,30 @@ impl Device {
         self.children.as_ref().map_or(false, |v| !v.is_empty())
     }
 
+    // Find first mounted device along with mountpoints
+    pub(crate) fn find_first_mounted(&self) -> Option<(&Device, String)> {
+        let output = Command::new("findmnt")
```

Comment:
So we already have code in `mount.rs` that's wrapping `findmnt`. An overall challenge here is because the install code is in a mount namespace, we won't see the device as mounted to start when we scan the block device.

That's why the `mountpoints` property that we can get from lsblk will be empty.

I think what we want here is a new api in `mount.rs` like this:

```
pub(crate) fn is_mounted_in_pid_mountns(device: &Device, pid: rustix::process::Pid) -> Result<bool>` or so.

Then we traverse the the device and run `findmnt -N <pid> --source <device>` or so

Reply:
So the `bootc install to-disk` code is in a mount namespace, which results in the device mountpoints being hidden from `findmnt`? 

For your suggestion, what process am I getting the pid from? Is that from the disk, since I'm feeding it into `findmnt` as the task id then?

Just trying to wrap my head around all this stuff, I appreciate the quick feedback a ton!

Reply:
> For your suggestion, what process am I getting the pid from?

Pid 1

Try playing with this from a root shell:

```
$ unshare -m
$ findmnt /tmp
$ mount --bind /var/tmp /tmp
$ findmnt /tmp
$ findmnt -N 1 /tmp
```

Note the second one (looking in the root mountns) didn't see the bind. Press ctrl-d to exit the `unshare` bash shell which will also tear down the mount.

Reply:
I think I have a better idea of it now, Joseph helped me walk through it yesterday too. My basic understanding is that `bootc install to-disk`  can't see the same pids (and hence the mountpoints) that the podman container can see, so I need to bridge the gap and allow install to-disk to see the mountpoints podman can. Then I can run `findmnt` with that specified pid and check for the mountpoints off of it

Reply:
I'll experiment with those commands you listed too, thank you